test/unit/eigensolver/mc/test_backtransformation.cpp

```diff
@@ -151,6 +161,104 @@ void testBacktransformationEigenv(SizeType m, SizeType n, SizeType mb, SizeType
   CHECK_MATRIX_NEAR(result, mat_c, error, error);
 }
 
+template <class T>
+void testBacktransformationEigenv(comm::CommunicatorGrid grid, SizeType m, SizeType n, SizeType mb,
+                                  SizeType nb) {
+  comm::Index2D src_rank_index(std::max(0, grid.size().rows() - 1), std::min(1, grid.size().cols() - 1));
+
+  LocalElementSize sizeC(m, n);
+  TileElementSize blockSizeC(mb, nb);
+  GlobalElementSize szC = globalTestSize(sizeC);
+  Distribution distrC(szC, blockSizeC, grid.size(), grid.rank(), src_rank_index);
+  Matrix<T, Device::CPU> mat_c(std::move(distrC));
+  dlaf::matrix::util::set_random(mat_c);
+
+  LocalElementSize sizeV(m, m);
+  TileElementSize blockSizeV(mb, mb);
+  GlobalElementSize szV = globalTestSize(sizeV);
+  Distribution distrV(szV, blockSizeV, grid.size(), grid.rank(), src_rank_index);
+  Matrix<T, Device::CPU> mat_v(std::move(distrV));
+  dlaf::matrix::util::set_random(mat_v);
+
+  if (m == 0 || n == 0)
+    return;
+
+  const SizeType nr_reflector = std::max(static_cast<SizeType>(0), m - mb - 1);
+
+  // Copy matrices locally
+  auto mat_c_loc = dlaf::matrix::test::allGather<T>(lapack::MatrixType::General, mat_c, grid);
```

Comment:
Since this is not going to be altered in any way, it can be gathered as a `const T`.